### PR TITLE
Annotate task bundles with version

### PR DESCRIPTION
**Pull request by tkdchen**
[STONEBLD-2821](https://issues.redhat.com//browse/STONEBLD-2821)

# Before you complete this pull request ...

Look for any open pull requests in the repository with the title "e2e-tests update" and 
see if there are recent e2e-tests updates that will be applicable to your change.

### CODE REVIEW

File: hack/build-and-push.sh [hunk 1]
```diff
@@ -172,6 +172,8 @@ do
       ANNOTATIONS+=("org.opencontainers.image.source=${VCS_URL}")
       ANNOTATIONS+=("org.opencontainers.image.revision=${VCS_REF}")
       ANNOTATIONS+=("org.opencontainers.image.url=${VCS_URL}/tree/${VCS_REF}/${task_dir}")
+      concrete_task_version=$(yq '.metadata.labels."app.kubernetes.io/version"' "$prepared_task_file")
```
Comment (mmorhun) [inline hack/build-and-push.sh:175]: Please use just `task_version`

Comment (tkdchen) [reply hack/build-and-push.sh:175]: Variable `task_version` is already used in the `read` command to reference the version from task file path.

File: hack/build-and-push.sh [hunk 2]
```diff
@@ -172,6 +172,8 @@ do
...
+      ANNOTATIONS+=("org.opencontainers.image.version=${concrete_task_version}")
```
Comment (chmeliik) [inline hack/build-and-push.sh:176]: Not all tasks have the version label I think. Will this do something sensible for those that don't (probably set the annotation to an empty string?)

Comment (tkdchen) [reply hack/build-and-push.sh:177]: Updated. PTAL.

**Review by chmeliik — APPROVED** [review]

**Pull request merged**